### PR TITLE
feat(appeals/web): publish valid reps functionality

### PR DESCRIPTION
**Pull request by emil-placheta**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/web/src/server/applications/case/representations/publish-valid-representations/publish-valid-reps.controller.js [hunk 1]
```diff
@@ -19,9 +22,30 @@ export const getPublishValidRepsController = async (req, res) => {
...
+	const payload = getPublishableRepsPayload(session, items)
+
+	const response = await publishPublishableReps(caseId, payload);
+
+	if(response && response.publishedRepIds.length === payload.json.representationIds.length) {
+		logger.info(`Successfully published ${response.publishedRepIds} representations for case ${caseId}`);
+		res.redirect(`${serviceUrl}/case/${caseId}/${representationsUrl}?published=${response.publishedRepIds.length}`);
+	} else {
+		res.redirect(`${serviceUrl}/case/${caseId}/${representationsUrl}`);
+	}
```
Comment (emil-placheta) [inline apps/web/src/server/applications/case/representations/publish-valid-representations/publish-valid-reps.controller.js:50]: to be refactored after the success banner is added

File: apps/web/src/server/applications/case/representations/publish-valid-representations/_utils/getPublishableRepsPayload.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import * as authSession from '../../../../../app/auth/auth-session.service.js';
```
[the remark this answers is not in the document]
Comment (emil-placheta) [reply apps/web/src/server/applications/case/representations/publish-valid-representations/_utils/getPublishableRepsPayload.js:1]: argh, I don't even know why I did that, will change

File: apps/web/src/server/applications/case/representations/publish-valid-representations/publish-valid-reps.router.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
 import { Router as createRouter } from 'express';
 import asyncRoute from '../../../../lib/async-route.js';
-import { getPublishValidRepsController } from './publish-valid-reps.controller.js';
+import { getPublishValidRepsController, postPublishValidRepsController } from './publish-valid-reps.controller.js';
 
 const publishValidRepresentationsRouter = createRouter({ mergeParams: true });
 
 publishValidRepresentationsRouter
 	.route('/publish-valid-representations')
```
[the remark this answers is not in the document]
Comment (emil-placheta) [reply apps/web/src/server/applications/case/representations/publish-valid-representations/publish-valid-reps.router.js:8]: good point

File: apps/web/src/server/applications/case/representations/publish-valid-representations/publish-valid-reps.controller.js [hunk 4]
```diff
@@ -19,9 +22,30 @@ export const getPublishValidRepsController = async (req, res) => {
 	const { itemCount: publishableRepsCount } = await getPublishableReps(caseId);
 
 	res.render(view, {
```
Comment (christopher-semmens) [inline apps/web/src/server/applications/case/representations/publish-valid-representations/publish-valid-reps.controller.js:24]: Return before res.render

**Review by christopher-semmens — APPROVED** [review]

**Pull request merged**